discovery_engine_core/web-api2/src/server.rs

```diff
@@ -67,12 +67,14 @@ where
         let addr = config.net.bind_to;
         init_tracing(config.as_ref());
 
+        let json_config = JsonConfig::default().limit(config.net.max_body_size.try_into().unwrap());
```

Comment:
Looks good through given that `max_body_size` only exist to be used in the config it is probably better to change the type of `max_body_size` to match the one `limit` expects as a small improvement.

---

Not specific to this situation but I personally prefer to use `.try_into().unwrap_or(usize::MAX)` i.e. max-clamp the value in situations where:

- you can't change the input (we can change it here)
- it only fails with absurdly to high values
- there is no major problem with using a different value 

through the last point is often not simple to determine and doing it in some situations could not only be confusion for someone later down the line but also might even be a security vulnerability. You could say it's the choice between  "going on with a slightly changed value because crashing is bad" or "aborting because a changed value could have unexpected consequences".

Either way in this case we can just change the input (`max_body_size`) to match the integer type we need and avoid any problematic decisions entirely.

Reply:
thanks for explaining. here is the new change a104f45d5f9cc58d363b3e88df42103c34c0444f